File: src/RunTests/JavaScript/messages.js
/**
 * Log a message with the details of the test that failed
 * @param {string} input - A string with the raw input for the current test
 * @param {string} expectedOutput - A string with the raw expected output from the current test
 * @param {*} output - A variable with the actual results for the current tests
 */
const failedTest = function(input, expectedOutput, output){
  console.log('=====================================')
  console.log('\nTest Failed!')
  console.log('Input: ', input)
  console.log('Output: ', output)
  console.log('Expected Output: ', expectedOutput)
  console.log('\n')
}

/**
 * Log a message with the details of the executed tests
 * @param {number} totalTests - The total number of tests provided
 * @param {number} fails - The number of tests that failed
 */
const results = function(totalTests, fails){
  const success = totalTests - fails
  const result = fails === 0 ? 'Good Job!' : 'Keep Trying'
  console.log('=====================================')
  console.log('\nFinished running all tests.')
  console.log(`${fails} failed test(s).`)
  console.log(`${success} successful test(s).`)
  console.log('=====================================')
  console.log(`[${success}/${totalTests}] ${result}`)
}

module.exports = {
  failedTest,
  results
}

File: src/RunTests/Python/messages.py
"""
Print a message with the details of the failed test
:param expected_input: The input for this test, as read from the test file
:param expected_output: The output for this test, as read from the test file 
:param actual_output: The actual output for this test, result of executing the given function
:return: void
"""
def print_fail_test_message( expected_input, expected_output, actual_output ):
  print('=====================================')
  print('\nTest Failed!')
  print('Input:', expected_input)
  print('Output:', actual_output)
  print('Expected Output: {}\n'.format(expected_output))

"""
Print a message with the results of the tests
:param total_tests: The total number of test cases
:param fails: The number of failed test cases
:return: void
"""
def print_results_message( total_tests, fails ):
  success = total_tests - fails
  result = 'Good Job!' if fails is 0 else 'Keep Trying'
  print('=====================================')
  print('\nFinished running all tests.')
  print(fails, 'failed test(s).')
  print(success, 'successful test(s).')
  print('\n=====================================\n')
  print('[{}/{}] {}\n'.format( success, total_tests, result ))

